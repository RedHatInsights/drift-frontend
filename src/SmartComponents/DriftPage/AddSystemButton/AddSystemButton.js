import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@patternfly/react-core';
import { useDispatch } from 'react-redux';

import { addSystemModalActions } from '../../AddSystemModal/redux';

const AddSystemButton = ({ loading, isToolbar }) => {
    const dispatch = useDispatch();

    const handleButtonClick = () => {
        dispatch(addSystemModalActions.toggleAddSystemModal());
    };

    return (
        <Button
            variant='primary'
            onClick={ () => handleButtonClick() }
            isDisabled={ loading }
            ouiaId={ isToolbar ? 'add-to-comparison' : 'add-to-comparison-empty' }
            data-testid='add-to-comparison-button'
        >
            { isToolbar ? 'Add to comparison' : 'Add systems or baselines' }
        </Button>
    );
};

AddSystemButton.propTypes = {
    toggleAddSystemModal: PropTypes.func,
    isToolbar: PropTypes.bool,
    loading: PropTypes.bool
};
export default AddSystemButton;
