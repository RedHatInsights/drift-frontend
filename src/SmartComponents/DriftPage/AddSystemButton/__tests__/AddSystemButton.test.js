import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import AddSystemButton from '../AddSystemButton';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

describe('add system button', () => {
    const mockStore = configureStore();
    let store;

    beforeEach(() =>
        store = mockStore({})
    );

    it('should render correctly', async () => {
        const button = shallow(<Provider store={ store }><AddSystemButton /></Provider>);
        expect(toJson(button)).toMatchSnapshot();
    });
});
