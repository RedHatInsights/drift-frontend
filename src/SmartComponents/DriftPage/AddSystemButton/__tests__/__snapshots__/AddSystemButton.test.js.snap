// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add system button should render correctly 1`] = `
<Button
  isDisabled={false}
  variant="primary"
>
  Add to comparison
</Button>
`;
